janitor/utils.py

```diff
@@ -405,3 +406,227 @@ def _wrapped(x, *args, **kwargs):
             return return_val
 
     return _wrapped
+
+
+def _check_instance(entry: Dict):
+    """
+    Function to check instances in the expand_grid function.
+
+    This checks if entry is a dictionary,
+    checks the instance of value in key:value pairs in entry,
+    and makes changes to other types as deemed necessary. 
+
+    Additionally, type-specific errors are raised
+    if unsupported data types are passed in as values 
+    in the entry dictionary.
+
+    How each type is handled, and their associated exceptions,
+    are pretty clear from the code.
+    """
+    # dictionary should not be empty
+    if not entry:
+        raise ValueError("passed dictionary cannot be empty")
+    # If it is a NoneType, number, Boolean, or string,
+    # then wrap in a list
+    entry = {
+        key: [value]
+        if isinstance(value, (type(None), int, float, bool, str))
+        else value
+        for key, value in entry.items()
+    }
+
+    # Convert to list if value is a set|tuple|range
+    entry = {
+        key: list(value) if isinstance(value, (set, tuple, range)) else value
+        for key, value in entry.items()
+    }
+
+    # collect dataframes here
+    dfs = []
+
+    # collect non dataframes here, proper dicts
+    dicts = {}
+
+    for key, value in entry.items():
+
+        # exclude dicts:
+        if isinstance(value, dict):
+            raise TypeError("Nested dictionaries are not allowed")
+
+        # process arrays
+        if isinstance(value, np.ndarray):
+            if value.size == 0:
+                raise ValueError("array cannot be empty")
+            elif value.ndim == 1:
```

Comment:
Following DeepSource's recommendation:

```suggestion
            if value.ndim == 1:
```

Reference: https://deepsource.io/gh/ericmjl/pyjanitor/run/a1eee997-ac46-451d-b916-3393713ba9eb/python/PYL-R1720